.gitmodules

```diff
@@ -1,3 +1,9 @@
 [submodule "include/boost/compute"]
 	path = compute
 	url = https://github.com/boostorg/compute
+[submodule "external_libs/fmt"]
+	path = external_libs/fmt
+	url = https://github.com/fmtlib/fmt.git
+[submodule "external_libs/fast_double_parser"]
+	path = external_libs/fast_double_parser
+	url = https://github.com/lemire/fast_double_parser.git
```

Comment:
It seems there submodules are necessary for installing LightGBM from source, I think we should update the related document, to add ``--recursive`` in git clone commands.

Reply:
I believe we already ask users to add `--recursive` flag during cloning due to `boostorg/compute` module.
https://lightgbm.readthedocs.io/en/latest/Installation-Guide.html?highlight=recursive

Reply:
great!